tests/testsuite/update.rs

```diff
@@ -1637,3 +1639,429 @@ fn update_with_missing_feature() {
         )
         .run();
 }
+
+#[cargo_test]
+fn update_breaking_unstable() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name  =  "foo"
+                version  =  "0.0.1"
+                edition  =  "2015"
+                authors  =  []
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("update --breaking")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_status(1)
+        .with_stderr_contains("[ERROR] unexpected argument '--breaking' found")
+        .run();
+}
+
+#[cargo_test]
+fn update_breaking_dry_run() {
+    Package::new("incompatible", "1.0.0").publish();
+    Package::new("ws", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                # Check if formatting is preserved. Nothing here should change, due to dry-run.
+
+                [workspace]
+                members  =  ["foo"]
+
+                [workspace.dependencies]
+                ws  =  "1.0"  # Preserve formatting
+            "#,
+        )
+        .file(
+            "foo/Cargo.toml",
+            r#"
+                # Check if formatting is preserved. Nothing here should change, due to dry-run.
+
+                [package]
+                name  =  "foo"
+                version  =  "0.0.1"
+                edition  =  "2015"
+                authors  =  []
+
+                [dependencies]
+                incompatible  =  "1.0"  # Preserve formatting
+                ws.workspace  =  true  # Preserve formatting
+            "#,
+        )
+        .file("foo/src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+
+    Package::new("incompatible", "1.0.1").publish();
+    Package::new("ws", "1.0.1").publish();
+
+    Package::new("incompatible", "2.0.0").publish();
+    Package::new("ws", "2.0.0").publish();
+
+    p.cargo("update --dry-run --breaking")
+        .with_status(1)
+        .with_stderr_contains("[ERROR] unexpected argument '--breaking' found")
+        .run();
+
+    let root_manifest = p.read_file("Cargo.toml");
+    assert_e2e().eq(
+        &root_manifest,
+        str![[r#"
+
+                # Check if formatting is preserved. Nothing here should change, due to dry-run.
```

Comment:
nit: for something like this, I'd make assign the manifest into a variable and use that with `file` and with `eq!()`.  This makes the intent clearer that it is to be left unchanged.

Reply:
Fixed.